usecases/objects/get.go

```diff
@@ -76,17 +72,29 @@ func (m *Manager) GetObjects(ctx context.Context, principal *models.Principal,
 
 	m.metrics.GetObjectInc()
 	defer m.metrics.GetObjectDec()
-	return m.getObjectsFromRepo(ctx, offset, limit, sort, order, after, addl, tenant)
-}
 
-func (m *Manager) GetObjectsClass(ctx context.Context, principal *models.Principal,
-	id strfmt.UUID,
-) (*models.Class, error) {
-	err := m.authorizer.Authorize(principal, authorization.READ, authorization.Objects("", "", id))
+	objects, err := m.getObjectsFromRepo(ctx, offset, limit, sort, order, after, addl, tenant)
 	if err != nil {
 		return nil, err
 	}
 
+	// Filter objects based on authorization
+	resourceFilter := filter.New[*models.Object](m.authorizer, m.config.Config)
+	filteredObjects := resourceFilter.Filter(
+		principal,
+		objects,
+		authorization.READ,
+		func(obj *models.Object) []string {
+			return []string{authorization.Objects(obj.Class, tenant, obj.ID)}
+		},
+	)
+
+	return filteredObjects, nil
+}
+
+func (m *Manager) GetObjectsClass(ctx context.Context, principal *models.Principal,
+	id strfmt.UUID,
+) (*models.Class, error) {
```

Comment:
I think this only gets the collection config of one object, so I don't think we should filter here. 
I honestly didn't even know this endpoint existed or where it is used, but if I understand it right it does:
- searches all collections for an object with the given ID
  - if multiple picks a random (?) one
- returns the collection config for that one

I'd just require Collection("*") for this one

Reply:
it would still limit because it's used as search and to not error unauthz it would scope the results, to be consistent behaviour at least for REST PoV , wdyt if we keep it filtered, i think it's used in deperecated path and need deletion at some point

Reply:
But what exactly are we filtering here? I think this can only ever be one object, but to even get that far, we would have to search through all existing collections.

Reply:
Could you look up if this is even used? I don't remember this endpoint, so it might not be reachable

Reply:
it's used in deprecated path https://github.com/weaviate/weaviate/blob/main/adapters/handlers/rest/handlers_objects.go#L163

Reply:
> it's used in deprecated path

ok!

But I still don't understand why we would need filtering here? 

We are doing the following:

- searches all collections for an object with the given ID
  - if multiple picks a random (?) one

Now you want to filter, but for what? You either have permission for that object or not?

And further down in the stack, I _think_ we check for permission for that object anyways:
usecases/objects/get.go::GetObject

So I _think_ we do not need any authZ checks in here.

(The whole handler does not make much sense to me, GetObjectsClass seems to return the collection config for the randomly choosen object, but that is then only used to set some additional parameters, no idea what's the point of that🤷)